# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpTransactionChild.cpp
# Commit: 8e7c97a37073
# Full Hash: 8e7c97a37073a14ba832e2f5c27f2c11b7787ef5
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2023-03-15 16:26:49
# Regressor Bug: 1805372
# File Overlap Count: 1
# Description:
#   Bug 1805372 - Pass effectiveTRRMode to nsHttpChannel from socketTransport/DnsAndConnectSocket r=necko-reviewers,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D164539
# ==============================================================================

diff -r 5abfd5ad2e35 -r 8e7c97a37073 netwerk/protocol/http/HttpTransactionChild.cpp
--- a/netwerk/protocol/http/HttpTransactionChild.cpp	Wed Mar 15 12:41:26 2023 +0000
+++ b/netwerk/protocol/http/HttpTransactionChild.cpp	Wed Mar 15 13:00:46 2023 +0000
@@ -470,12 +470,25 @@
   int32_t proxyConnectResponseCode =
       mTransaction->GetProxyConnectResponseCode();
 
+  nsIRequest::TRRMode mode = nsIRequest::TRR_DEFAULT_MODE;
+  TRRSkippedReason reason = nsITRRSkipReason::TRR_UNSET;
+  {
+    NetAddr selfAddr;
+    NetAddr peerAddr;
+    bool isTrr = false;
+    bool echConfigUsed = false;
+    if (mTransaction) {
+      mTransaction->GetNetworkAddresses(selfAddr, peerAddr, isTrr, mode, reason,
+                                        echConfigUsed);
+    }
+  }
+
   Unused << SendOnStartRequest(
       status, optionalHead, securityInfo, mTransaction->ProxyConnectFailed(),
       ToTimingStructArgs(mTransaction->Timings()), proxyConnectResponseCode,
       dataForSniffer, optionalAltSvcUsed, !!mDataBridgeParent,
       mTransaction->TakeRestartedState(), mTransaction->HTTPSSVCReceivedStage(),
-      mTransaction->GetSupportsHTTP3());
+      mTransaction->GetSupportsHTTP3(), mode, reason);
   return NS_OK;
 }
 
@@ -582,8 +595,10 @@
     NetAddr peerAddr;
     bool isTrr = false;
     bool echConfigUsed = false;
+    nsIRequest::TRRMode mode = nsIRequest::TRR_DEFAULT_MODE;
+    TRRSkippedReason reason = nsITRRSkipReason::TRR_UNSET;
     if (mTransaction) {
-      mTransaction->GetNetworkAddresses(selfAddr, peerAddr, isTrr,
+      mTransaction->GetNetworkAddresses(selfAddr, peerAddr, isTrr, mode, reason,
                                         echConfigUsed);
     } else {
       nsCOMPtr<nsISocketTransport> socketTransport =
@@ -592,10 +607,12 @@
         socketTransport->GetSelfAddr(&selfAddr);
         socketTransport->GetPeerAddr(&peerAddr);
         socketTransport->ResolvedByTRR(&isTrr);
+        socketTransport->GetEffectiveTRRMode(&mode);
+        socketTransport->GetTrrSkipReason(&reason);
         socketTransport->GetEchConfigUsed(&echConfigUsed);
       }
     }
-    arg.emplace(selfAddr, peerAddr, isTrr, echConfigUsed);
+    arg.emplace(selfAddr, peerAddr, isTrr, mode, reason, echConfigUsed);
   }
 
   Unused << SendOnTransportStatus(aStatus, aProgress, aProgressMax, arg);